README.md
# GeckoMGV
Multigenome viewer in Django and JS - University of Málaga

## Installation

To run GeckoMGV on your computer you should install first some dependencies, starting with python and virtualenv, so:

* `apt-get install python virtualenv`
* `git clone https://github.com/Sergiodiaz53/GeckoMGV`

Then, I recommend to create a virtualenv for this project:

* `virutalenv GeckoMGV `

And activate it:

* `cd GeckoMGV/`
* `source bin/activate`

Now we should be working inside our virtualenv, so we are going to install django and its plugins

* pip install django==179 django-forms-builder django_extensions

We are going to set Django up:

* `python manage.py migrate`
* `python manage.py createsuperuser`

Finally, we should be able to run GeckoMGV with:

* `python manage.py runserver`

*· You can modify the ip address and the port using `python manage.py runserver 0.0.0.0:8000`*

*· Installing Werkzeug (`pip install Werkzeug`) you can use `python manage.py runserver_plus` for debug details*

For extra documentation about how to register a service, user guide for GeckoMGV, modules description... you can visit our server [PISTACHO](https://pistacho.ac.uma.es)


CHANGELOG
30/09/2016
- Frags info now working with Slickgrid
-

--------------------------------------------------

09/03/2016
- Services execution from index improvements.
    - Alerts added to service execution callback.
    - Execution in background improved.


08/03/2016
- CreateFile now accept filename and file format.
    - Some bugs fixed in fileSystem

07/03/2016
- Manual filtering

01/03/2016
- Matrix visualization improvements.

27/02/2016
- Methods to generate Matrix added.
- Identify filter line added to Matrix.

25/02/2016
- Matrix in D3.js Added

15/02/2016
- Methods to call services from the index added (TESTING)

27/01/2016
-B16 (control of map and horizontal layers on zoom) solved
-Added upload to server option to upload selected frags.

11/01/2016
- New interface implemented
- Some bugs fixed

15/12/2015
-Added "back to previous zoom" button
-Bugs related to info tables solved.

17/11/2015
- Fixed some bugs

10/11/2015
- Fixed map bug
- Loading icon added
- Papaparse use web-workers now
- Annotations paint modified
- Enter to search on Annotation

04/11/2015
-A search will directly paint crossed lines to their possition on the sequence

03/11/2015
-Annotation search adviable displaying possibility of undo a filter or ad a new one

28/10/2015
- Changed path variables.
- Updated 'executeServices()' from Serverside
- Minor improvements in the interface

14/10/2015
- Added filename field to the userFile model
- Fixed directory dependencies between production/development
- Minor changes in the fileSystem app
- Minor changes in the settings.py for improve security
- Now we load csv files through Papaparse

07/10/2015
- Generate results files with the same name fixed

06/10/2015
- Preview_View shows the content of a file after execution and also from the fileManager_View
- LocalStorage deprecated for 'lines'
- Modify results filename is allowed now
- Now you can filter annotations by their contents writing on the search box and clicking on the search button.

05/10/2015
Save files after execution - alpha version

30/09/2015
-Displayed data can be downloaded as .csv by clicking on save
-Displayed data can change between selected and on the current view by clicking on selected

29/09/2015
-Frags remain in red while they're selected.
-Canvas info and tittle will vary depending on number of frags, also the popover from the info have as title the sequences compared.

28/09/2015
- Added transparency to Horizontal view
- Main panels of index are now collapsible
- Added CSB2CSV
- Single frags can be selectec and unselected by a single click while holdingh Shift

24/09/2015
- From now only filtered frags will be selected when using selection tool.
- Added multiple selection tool, it's possible to select many frags many times by selecting other areas.
- Fixed ordering parameters issue in Django. Invoking services now working correctly

23/09/2015
- Modified the way CSB & Frag info and annotations are showed, Now are jQueryUI dialogs with bootstrap looklike (jQueryUI plugin).
- jQueryUI updated.
- Key event SHIFT now can select an area of frags.
- Modified key event to force squared area, now either Ctrl or Alt.

21/09/2015
- Key event SHIFT added for block zoom and select frags.
- Key event AltGr to force squared area selection (Zoom & Selection).
- Added 'load Mauven' to load and convert Mauven files into GeckoMGV.



scripts/forms.py
from django import forms
from .models import models
from django.contrib.auth import get_user_model
from scripts.models import *
from fileSystem.views import *

User = get_user_model()

class RegisterService(forms.ModelForm):
    class Meta:
        fields = ["name", "exeName", "path", "help", "form"]
        model = Script

class ExecuteService(forms.Form):
    class Meta:
        fields = ["name", "args"]

class drawMSAComp(forms.Form):
    seq1=forms.CharField(label='Sequence2', max_length=100)
    seq2=forms.CharField(label='Sequence1', max_length=100)


class kmersForm(forms.Form):
    parameter1 = forms.ChoiceField(label="Filename", widget=forms.Select(attrs={'class':'selector','id': 'Filename'}))
    parameter2 = forms.CharField(label='K', max_length=1, widget=forms.TextInput(attrs={'id': 'K'}))
    parameter3 = forms.CharField(label='fullOut', max_length=1, widget=forms.NumberInput(attrs={'id': 'fullOut'}))


    def __init__(self, *args, **kwargs):
        self.user = kwargs.pop('user', None)
        self.request = kwargs.pop('request', None)
        super(kmersForm, self).__init__(*args, **kwargs)
        self.fields['parameter1'] = forms.ChoiceField(choices=[(file.file.name, file.filename)  for file in userFile.objects.filter(user=self.user)], label="Filename", widget=forms.Select(attrs={'class':'selector','id': 'Filename'}))

class csb2csvForm(forms.Form):
    parameter1 = forms.ChoiceField(label="CSB file", widget=forms.Select(attrs={'class':'selector','id': 'CSBfile'}))
    parameter2 = forms.ChoiceField(label="Master file", widget=forms.Select(attrs={'class':'selector','id': 'Masterfile'}))


    def __init__(self, *args, **kwargs):
        self.user = kwargs.pop('user', None)
        self.request = kwargs.pop('request', None)
        super(csb2csvForm, self).__init__(*args, **kwargs)
        self.fields['parameter1'] = forms.ChoiceField(choices=[(file.file.name, file.filename)  for file in userFile.objects.filter(user=self.user)], label="CSB file", widget=forms.Select(attrs={'class':'selector','id': 'CSBfile'}))
        self.fields['parameter2'] = forms.ChoiceField(choices=[(file.file.name, file.filename)  for file in userFile.objects.filter(user=self.user)], label="Master file", widget=forms.Select(attrs={'class':'selector','id': 'Masterfile'}))

class clustalForm(forms.Form):
    parameter1 = forms.ChoiceField(label="Multifasta file", widget=forms.Select(attrs={'class':'selector','id': 'multifile'}))


    def __init__(self, *args, **kwargs):
        self.user = kwargs.pop('user', None)
        self.request = kwargs.pop('request', None)
        super(clustalForm, self).__init__(*args, **kwargs)
        self.fields['parameter1'] = forms.ChoiceField(choices=[(file.file.name, file.filename)  for file in userFile.objects.filter(user=self.user)], label="Multifasta file", widget=forms.Select(attrs={'class':'selector','id': 'multifile'}))

class extractSeqFromFragsForm(forms.Form):
    parameter1 = forms.ChoiceField(label="frags file", widget=forms.Select(attrs={'class':'selector','id': 'fragsFile'}))
    parameter2 = forms.ChoiceField(label="X Fasta file", widget=forms.Select(attrs={'class':'selector','id': 'xFastaFile'}))
    parameter3 = forms.ChoiceField(label="Y Fasta file", widget=forms.Select(attrs={'class':'selector','id': 'yFastaFile'}))
    parameter4 = forms.ChoiceField(label="Y-reversed Fasta file", widget=forms.Select(attrs={'class':'selector','id': 'yReversedFastaFile'}))
    parameter5 = forms.CharField(label="Output FragFile", widget=forms.TextInput(attrs={'class':'file','id': 'outputFragFile'}))
    parameter6 = forms.CharField(label='Block', max_length=4, widget=forms.TextInput(attrs={'id': 'block', 'placeholder':'e.g: 0'}))

    def __init__(self, *args, **kwargs):
        self.user = kwargs.pop('user', None)
        self.request = kwargs.pop('request', None)
        super(extractSeqFromFragsForm, self).__init__(*args, **kwargs)
        self.fields['parameter1'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="frags file", widget=forms.Select(attrs={'class':'selector','id': 'fragsFile'}))
        self.fields['parameter2'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="X Fasta file", widget=forms.Select(attrs={'class':'selector','id': 'xFastaFile'}))
        self.fields['parameter3'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Y Fasta file", widget=forms.Select(attrs={'class':'selector','id': 'yFastaFile'}))
        self.fields['parameter4'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Y-reversed Fasta file", widget=forms.Select(attrs={'class':'selector','id': 'yReversedFastaFile'}))

        #self.fields['parameter5'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Output FragFile", widget=forms.Select(attrs={'class':'selector','id': 'outputFragFile'}))

class reverseComplementForm(forms.Form):
    parameter1 = forms.ChoiceField(label="Fasta to reverse", widget=forms.Select(attrs={'class':'selector','id': 'fastaToReverse'}))
    parameter2 = forms.CharField(label="Output file", widget=forms.TextInput(attrs={'class':'file','id': 'outputFile'}))

    def __init__(self, *args, **kwargs):
        self.user = kwargs.pop('user', None)
        self.request = kwargs.pop('request', None)
        super(reverseComplementForm, self).__init__(*args, **kwargs)
        self.fields['parameter1'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Fasta to reverse", widget=forms.Select(attrs={'class':'selector','id': 'fastaToReverse'}))
        #self.fields['parameter2'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Output file", widget=forms.Select(attrs={'class':'selector','id': 'outputFile'}))

class extractOverlappedForm(forms.Form):
    parameter1 = forms.ChoiceField(label="Master file", widget=forms.Select(attrs={'class':'selector','id': 'masterFile'}))
    parameter2 = forms.CharField(label="Not overlapped output file", widget=forms.TextInput(attrs={'class':'file','id': 'nonOverlapped'}))
    parameter3 = forms.CharField(label="Overlapped output file", widget=forms.TextInput(attrs={'class':'file','id': 'overlapped'}))
    parameter4 = forms.ChoiceField(label="Sequence x", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
    parameter5 = forms.ChoiceField(label="Sequence y", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))

    def __init__(self, *args, **kwargs):
        self.user = kwargs.pop('user', None)
        self.request = kwargs.pop('request', None)
        super(extractOverlappedForm, self).__init__(*args, **kwargs)
        self.fields['parameter1'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Master file", widget=forms.Select(attrs={'class':'selector','id': 'masterFile'}))

        #self.fields['parameter3'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Not Overlapped output file", widget=forms.Select(attrs={'class':'selector','id': 'nonOverlapped'}))
        #self.fields['parameter6'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Overlapped output file", widget=forms.Select(attrs={'class':'selector','id': 'overlapped'}))

        self.fields['parameter4'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence x", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
        self.fields['parameter5'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence y", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))

class connectFragsUpForm(forms.Form):
    parameter1 = forms.ChoiceField(label="Original frags file", widget=forms.Select(attrs={'class':'selector','id': 'originalFrags'}))
    parameter2 = forms.CharField(label="Output file", widget=forms.TextInput(attrs={'class':'file','id': 'outputFile'}))
    parameter3 = forms.ChoiceField(label="Sequence X FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
    parameter4 = forms.ChoiceField(label="Sequence Y FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))

    def __init__(self, *args, **kwargs):
        self.user = kwargs.pop('user', None)
        self.request = kwargs.pop('request', None)
        super(connectFragsUpForm, self).__init__(*args, **kwargs)
        self.fields['parameter1'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Original frags file", widget=forms.Select(attrs={'class':'selector','id': 'originalFrags'}))

        #self.fields['parameter2'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Output file", widget=forms.Select(attrs={'class':'selector','id': 'outputFile'}))

        self.fields['parameter3'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence X FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
        self.fields['parameter4'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence Y FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))

class GeckoCSB(forms.Form):
    parameter1 = forms.ChoiceField(label="Original frags file", widget=forms.Select(attrs={'class':'selector','id': 'originalFrags'}))
    parameter2 = forms.CharField(label="Overlapped file", widget=forms.TextInput(attrs={'class':'file','id': 'overFile'}))
    parameter5 = forms.CharField(label="CSB file", widget=forms.TextInput(attrs={'class':'file','id': 'csbFile'}))
    parameter6 = forms.CharField(label="Master file", widget=forms.TextInput(attrs={'class':'file','id': 'masterFile'}))
    parameter3 = forms.CharField(label="iGap", widget=forms.TextInput(attrs={'id': 'igap'}))
    parameter4 = forms.IntegerField(label='eGap', widget=forms.TextInput(attrs={'id': 'egap'}))
    parameter7 = forms.ChoiceField(label="Sequence X FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
    parameter8 = forms.ChoiceField(label="Sequence Y FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))

    def __init__(self, *args, **kwargs):
        self.user = kwargs.pop('user', None)
        self.request = kwargs.pop('request', None)
        super(GeckoCSB, self).__init__(*args, **kwargs)
        self.fields['parameter1'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Original frags file", widget=forms.Select(attrs={'class':'selector','id': 'originalFrags'}))
        self.fields['parameter7'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence X FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
        self.fields['parameter8'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence Y FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))

class refineOverlappedForm(forms.Form):
    parameter1 = forms.ChoiceField(label="Overlapped frags file", widget=forms.Select(attrs={'class':'selector','id': 'overFile'}))
    parameter2 = forms.ChoiceField(label="Not overlapped file", widget=forms.Select(attrs={'class':'selector','id': 'notFrags'}))
    parameter3 = forms.CharField(label="New Master frags file", widget=forms.TextInput(attrs={'class':'file','id': 'newMFile'}))
    parameter4 = forms.CharField(label="New overlapped frags file", widget=forms.TextInput(attrs={'class':'file','id': 'newOFrags'}))
    parameter5 = forms.CharField(label="New not overlapped frags file", widget=forms.TextInput(attrs={'class':'file','id': 'newNFile'}))
    parameter6 = forms.ChoiceField(label="Sequence X FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
    parameter7 = forms.ChoiceField(label="Sequence Y FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))

    def __init__(self, *args, **kwargs):
        self.user = kwargs.pop('user', None)
        self.request = kwargs.pop('request', None)
        super(refineOverlappedForm, self).__init__(*args, **kwargs)
        self.fields['parameter1'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Overlapped frags file", widget=forms.Select(attrs={'class':'selector','id': 'overFile'}))
        self.fields['parameter2'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Not overlapped file", widget=forms.Select(attrs={'class':'selector','id': 'notFrags'}))
        #self.fields['parameter4'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="New Master frags file", widget=forms.Select(attrs={'class':'selector','id': 'newMFile'}))
        #self.fields['parameter5'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="New overlapped frags file", widget=forms.Select(attrs={'class':'selector','id': 'newOFrags'}))
        #self.fields['parameter6'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="New not overlapped frags file", widget=forms.Select(attrs={'class':'selector','id': 'newNFile'}))
        self.fields['parameter6'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence X FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
        self.fields['parameter7'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence Y FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))

class joinCSBwithOverlappedForm(forms.Form):
    parameter1 = forms.ChoiceField(label="CSB file", widget=forms.Select(attrs={'class':'selector','id': 'csbin'}))
    parameter2 = forms.ChoiceField(label="Overlapped file", widget=forms.Select(attrs={'class':'selector','id': 'overin'}))
    parameter3 = forms.CharField(label="CSB output", widget=forms.TextInput(attrs={'class':'file','id': 'csbout'}))
    parameter4 = forms.CharField(label="New master file", widget=forms.TextInput(attrs={'class':'file','id': 'newMaster'}))
    parameter5 = forms.CharField(label="iGap", widget=forms.TextInput(attrs={'id': 'igap'}))
    parameter6 = forms.IntegerField(label='eGap', widget=forms.TextInput(attrs={'id': 'egap'}))
    parameter7 = forms.ChoiceField(label="Sequence X FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
    parameter8 = forms.ChoiceField(label="Sequence Y FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))

    def __init__(self, *args, **kwargs):
        self.user = kwargs.pop('user', None)
        self.request = kwargs.pop('request', None)
        super(joinCSBwithOverlappedForm, self).__init__(*args, **kwargs)
        self.fields['parameter1'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="CSB file", widget=forms.Select(attrs={'class':'selector','id': 'csbin'}))
        self.fields['parameter2'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Overlapped file", widget=forms.Select(attrs={'class':'selector','id': 'overin'}))
        self.fields['parameter7'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence X FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
        self.fields['parameter8'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence Y FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))

class classifyRepeatsForm(forms.Form):
    parameter4 = forms.CharField(label="Tandem repeats", widget=forms.TextInput(attrs={'class':'file','id': 'tandemFrags'}))
    parameter1 = forms.ChoiceField(label="Overlapped frags file", widget=forms.Select(attrs={'class':'selector','id': 'overFile'}))
    parameter2 = forms.ChoiceField(label="Not overlapped file", widget=forms.Select(attrs={'class':'selector','id': 'notFrags'}))
    parameter3 = forms.CharField(label="New Master frags file", widget=forms.TextInput(attrs={'class':'file','id': 'newMFile'}))
    parameter5 = forms.CharField(label="Interspersed repeats", widget=forms.TextInput(attrs={'class':'file','id': 'interFrags'}))
    parameter6 = forms.CharField(label="Duplications", widget=forms.TextInput(attrs={'class':'file','id': 'dupFile'}))
    parameter7 = forms.CharField(label="Not overlapped repeats", widget=forms.TextInput(attrs={'class':'file','id': 'notFile'}))
    parameter8 = forms.ChoiceField(label="Sequence X FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
    parameter9 = forms.ChoiceField(label="Sequence Y FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))

    def __init__(self, *args, **kwargs):
        self.user = kwargs.pop('user', None)
        self.request = kwargs.pop('request', None)
        super(classifyRepeatsForm, self).__init__(*args, **kwargs)

        #self.fields['parameter4'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Tandem repeats", widget=forms.Select(attrs={'class':'selector','id': 'tandemFrags'}))
        self.fields['parameter1'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Overlapped frags file", widget=forms.Select(attrs={'class':'selector','id': 'overFile'}))
        self.fields['parameter2'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Not overlapped file", widget=forms.Select(attrs={'class':'selector','id': 'notFrags'}))
        #self.fields['parameter3'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="New Master frags file", widget=forms.Select(attrs={'class':'selector','id': 'newMFile'}))
        #self.fields['parameter5'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Interspersed repeats", widget=forms.Select(attrs={'class':'selector','id': 'interFrags'}))
        #self.fields['parameter6'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Duplications", widget=forms.Select(attrs={'class':'selector','id': 'dupFile'}))
        #self.fields['parameter7'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Nor overlapped repeats", widget=forms.Select(attrs={'class':'selector','id': 'notFile'}))
        self.fields['parameter8'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence X FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
        self.fields['parameter9'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence Y FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))

class geckoForm(forms.Form):
    choice=(("2","8"),("3","12"),("4","16"),("5","20"),("6","24"),("7","28"),("8","32"))
    parameter1 = forms.ChoiceField(label="Sequence X FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
    parameter2 = forms.ChoiceField(label="Sequence Y FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))
    parameter3 = forms.IntegerField(label='min length', widget=forms.TextInput(attrs={'id': 'leng'}))
    parameter4 = forms.CharField(label='min similarity', max_length=3, widget=forms.TextInput(attrs={'id': 'sim'}))
    parameter5 = forms.ChoiceField(label='word length', choices=choice, widget=forms.Select(attrs={'class':'selector','id': 'Wleng'}))
    parameter6 = forms.CharField(label='fixed length', initial="1",widget = forms.HiddenInput())
    parameter7 = forms.CharField(label="CSV result", widget=forms.TextInput(attrs={'class':'file','id': 'csvout'}))

    def __init__(self, *args, **kwargs):
        self.user = kwargs.pop('user', None)
        self.request = kwargs.pop('request', None)
        super(geckoForm, self).__init__(*args, **kwargs)
        self.fields['parameter1'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence X FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqx'}))
        self.fields['parameter2'] = forms.ChoiceField(choices=[(file.file.name, file.filename) for file in userFile.objects.filter(user=self.user)], label="Sequence Y FASTA", widget=forms.Select(attrs={'class':'selector','id': 'seqy'}))
